lesson_6/edmond_sabou/functions2.py

```diff
@@ -0,0 +1,123 @@
+    ('HR', ['61', '66', '65', '68', '77', '77', '76', '82', '81']),
+    ('HU', ['63', '67', '70', '73', '76', '79', '82', '83', '86']),
+    ('IE', ['78', '81', '82', '82', '85', '87', '88', '89', '91']),
+    ('IS', ['93', '95', '96', '96', '0', '0', '98', '99', '98']),
+    ('IT', ['62', '63', '69', '73', '75', '79', '81', '84', '85']),
+    ('LT', ['60', '60', '65', '66', '68', '72', '75', '78', '82']),
+    ('LU', ['91', '93', '94', '96', '97', '97', '97', '93', '95']),
+    ('LV', ['64', '69', '72', '73', '76', '77', '79', '82', '85']),
+    ('ME', ['0', '55', '0', '0', '0', '0', '71', '72', '74']),
+    ('MK', ['0', '58', '65', '68', '69', '75', '74', '79', '82']),
+    ('MT', ['75', '77', '78', '80', '81', '81', '85', '84', '86']),
+    ('NL', ['94', '94', '95', '96', '96', '97', '98', '98', '98']),
+    ('NO', ['92', '93', '94', '93', '97', '97', '97', '96', '98']),
+    ('PL', ['67', '70', '72', '75', '76', '80', '82', '84', '87']),
+    ('PT', ['58', '61', '62', '65', '70', '74', '77', '79', '81']),
+    ('RO', ['47', '54', '58', '61', '68', '72', '76', '81', '84']),
+    ('RS', ['0', '0', '0', '0', '64', '0', '68', '73', '80']),
+    ('SE', ['91', '92', '93', '90', '91', '94', '95', '93', '96']),
+    ('SI', ['73', '74', '76', '77', '78', '78', '82', '87', '89']),
+    ('SK', ['71', '75', '78', '78', '79', '81', '81', '81', '82']),
+    ('TR', ['0', '47', '49', '60', '70', '76', '81', '84', '88']),
+    ('UK', ['83', '87', '88', '90', '91', '93', '94', '95', '96']),
+    ('XK', ['0', '0', '0', '0', '0', '0', '89', '93', '93']),
+]
+
+
+#######################################################################################################################
+def create():
+    new_dict = {country: data_list for country, data_list in raw_data}
+
+    return {
+        country: [
+            {"year": "2011", "coverage": data_list[0]}, {"year": "2012", "coverage": data_list[1]},
+            {"year": "2013", "coverage": data_list[2]}, {"year": "2014", "coverage": data_list[3]},
+            {"year": "2015", "coverage": data_list[4]}, {"year": "2016", "coverage": data_list[5]},
+            {"year": "2017", "coverage": data_list[6]}, {"year": "2018", "coverage": data_list[7]},
+            {"year": "2019", "coverage": data_list[8]}
+        ]
+        for country, data_list in new_dict.items()
+    }
+
+
+#######################################################################################################################
+#
+# get_year_data(data_set, "2019")
+#
+# >>> {'2019': [('Romania', 84), ('Germany', 95), ..., ('Latvia', 85)]}
+#
+def get_year_data(new_data_set, year):
+    #    print(type(new_data_set))
+    #    print(new_data_set)
+    new = {
+        year: [(country, index["coverage"])
+               for country in new_data_set.keys()
+               for index in new_data_set[country]
+               if index["year"] == str(year)]  # Q: De ce este nevoie sa pun str(year)??

```

Comment:
if you call the function get_year_data with year as string you don't have to convert it to string